userspace/libpman/src/lifecycle.c

```diff
@@ -28,6 +29,82 @@ int pman_open_probe() {
 	return 0;
 }
 
+int pman_prepare_progs_before_loading() {
+	char msg[MAX_ERROR_MESSAGE_LEN];
+	/*
+	 * Probe required features for each bpf program, as requested
+	 */
+	errno = 0;
+	for(int ev = 0; ev < PPM_EVENT_MAX; ev++) {
+		event_prog_t *progs = event_prog_table[ev];
+		int idx, chosen_idx = -1;
+		for(idx = 0; idx < MAX_FEATURE_CHECKS && progs[idx].name != NULL; idx++) {
+			bool should_disable = chosen_idx != -1;
+			if(!should_disable) {
+				if(progs[idx].feat > 0 &&
+				   libbpf_probe_bpf_helper(BPF_PROG_TYPE_RAW_TRACEPOINT, progs[idx].feat, NULL) ==
+				           0) {
+					snprintf(msg,
+					         MAX_ERROR_MESSAGE_LEN,
+					         "BPF program '%s' did not satisfy required feature [%d]",
+					         progs[idx].name,
+					         progs[idx].feat);
+					pman_print_msg(FALCOSECURITY_LOG_SEV_DEBUG, (const char *)msg);
+					// Required feature not present
+					should_disable = true;
+				} else {
+					// We satified requested feature
+					snprintf(msg,
+					         MAX_ERROR_MESSAGE_LEN,
+					         "BPF program '%s' satisfied required feature [%d]",
+					         progs[idx].name,
+					         progs[idx].feat);
+					pman_print_msg(FALCOSECURITY_LOG_SEV_DEBUG, (const char *)msg);
+					chosen_idx = idx;
```

Comment:
Could we just `break` out of the inner loop here and use `idx` directly in the next block? It would also mean we don't need `should_disable` because the final block in the loop will only be reached from the other branch of this if-else block.

Reply:
Mmmh if we break somewhere, we would lose the ability to disable all unused programs, right?

Reply:
You could disable the ones you skip over as you loop, then disable anything that might be after the chosen one on a separate loop, but either way is fine, I didn't realize it was meant to disable unused ones.